### PR TITLE
Added git-jira-branch.

### PR DESCRIPTION
**Pull request by martyphee**
This adds a JIRA integration.  Only jira branch is in this PR.

### CODE REVIEW

File: bin/git-jira-branch.rb [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    super
+    parse_args!(args)
+
+    if args.count < 1
+      show_and_exit usage
+    end
+
+    story_id = args[0]
+
+    # get PT info
```
Comment (pedrocunha) [inline bin/git-jira-branch.rb]: leftover here?

File: bin/git-jira-branch.rb [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    puts ' or Press CTRL-D to cancel'
+
+    branch_name = Readline.readline '  > ', false
+
+    if branch_name.nil?
+      log.warn "\nCancelled by user"
+      exit 2
+    end
+
+    puts 'Would you like to transition the issue?'
```
Comment (pedrocunha) [inline bin/git-jira-branch.rb]: Out of curiosity (for a non JIRA user here) what are the possible transitions? This is an interesting feature by the way

File: bin/git-jira-branch.rb [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    # create branch
+    `cd #{ENV['PWD']} && git checkout -b #{branch_name}`
+
+    # comment and start on PT
+    # issue.notes.create :text => "Created branch #{branch_name}"
+    # issue.update :current_state => 'started'
+
+    if transition
+      # Seems we need to do this in two updates.  JIRA complained if we tried to transition and update fields.
+      attrs = {
```
Comment (pedrocunha) [inline bin/git-jira-branch.rb]: I'm not sure if this is your style or not, but indentation doesn't seem consistent between variable assignments?

File: bin/git-jira-branch.rb [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+  end
+
+  def get_jira_info(story_id)
+    @username = `git config jira.username`.strip
+    if @username.empty?
+      puts Term::ANSIColor.yellow %Q{
+        Your branch appears to have a story ID,
+        but I don't know your JIRA username!
+        Please set it with:
+        $ git config [--global] jira.username <token>
```
Comment (pedrocunha) [inline bin/git-jira-branch.rb]: should it be `<username>` ?

File: bin/git-jira-branch.rb [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+      auth_type: :basic,
+      read_timeout: 120
+    }
+
+    client = JIRA::Client.new(options)
+    issue = client.Issue.find(story_id)
+
+    log.info '.'
+
+    log.info "Found story #{story_id} in '#{issue.fields['project']['name']}'"
```
Comment (pedrocunha) [inline bin/git-jira-branch.rb]: is JIRA stories or issues the terminology?

File: bin/git-jira-branch.rb (hunk 1 above)
Comment (martyphee) [inline bin/git-jira-branch.rb]: yeah

**Comment by pedrocunha** [comment]
Thanks for your contribution @martyphee looks good, I left a few questions :)

File: bin/git-jira-branch.rb (hunk 2 above)
Comment (martyphee) [inline bin/git-jira-branch.rb]: Their configurable.  That's why I present a list here.  There's no way of knowing what they are and I'm not sure if the order matters.

File: bin/git-jira-branch.rb (hunk 3 above)
Comment (martyphee) [inline bin/git-jira-branch.rb]: RubyMine formatting.  Let me fix that up.

File: bin/git-jira-branch.rb (hunk 4 above)
Comment (martyphee) [inline bin/git-jira-branch.rb]: Yes

File: bin/git-jira-branch.rb (hunk 2 above)
Comment (pedrocunha) [inline bin/git-jira-branch.rb]: Do you think it would be worth to at least order them alphabetically ?

File: bin/git-jira-branch.rb (hunk 5 above)
Comment (martyphee) [inline bin/git-jira-branch.rb]: JIRA refers to them as issues.

File: bin/git-jira-branch.rb (hunk 2 above)
Comment (martyphee) [inline bin/git-jira-branch.rb]: Not sure if that would be confusing is the numbers are out of order.  What do you think?

Comment (pedrocunha) [inline bin/git-jira-branch.rb]: Gotcha, i guess they will come ordered by that "id" so i think we should keep it. :+1:

File: bin/git-jira-branch.rb (hunk 3 above)
Comment (martyphee) [inline bin/git-jira-branch.rb]: How's this formatting?

```

File: bin/git-jira-branch.rb (hunk 5 above)
Comment (pedrocunha) [inline bin/git-jira-branch.rb]: I think if they are issues we should probably rename all "story" occurrences in this file to "issue" ? Or do you also use the word "story" to refer to JIRA issues ?

Comment (martyphee) [inline bin/git-jira-branch.rb]: I renamed them all to `issue` and `issue_id`

File: bin/git-jira-branch.rb (hunk 3 above)
Comment (pedrocunha) [inline bin/git-jira-branch.rb]: Yep, that works for me :+1:

**Comment by pedrocunha** [comment]
After you last commit did you have it go locally to see if everything is still working as expected? Sorry to ask you this, but I'm not really a JIRA user. The changes look good!

**Comment by martyphee** [comment]
It looks good. 

image

**Comment by pedrocunha** [comment]
Perfect :+1:

**Comment by martyphee** [comment]
I just renamed the file to drop the `.rb`

**Comment by pedrocunha** [comment]
I'll try to issue a release tomorrow morning, as there are a few things to be released if that's ok. I'll ping you here

**Comment by martyphee** [comment]
That's good.  If I get some time tonight I'm going to work on the create PR
version.

**Comment by mezis** [comment]
neato. thanks guys!

**Comment by mezis** [comment]
@pedrocunha I think you're gem owner already, so probably don't need me?

**Comment by pedrocunha** [comment]
Hey @martyphee do you mind trying out the new released version 1.0.1 to see if everything works for you? I did a bit of tidy up on your script (i missed some duplication on the code review). Thanks.

**Comment by martyphee** [comment]
I'll take a look.

**Comment by martyphee** [comment]
Can't seem to get it installed.

 BR ICE-85-add-locale-based-routing  ... | orderweb  gem install
git-whistles
ERROR:  While executing gem ... (Gem::Package::PathError)
    installing into parent path
/Users/martinphee/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/libexec/runner.rb
of
/Users/martinphee/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/git-whistles-1.0.1
is not allowed

**Comment by pedrocunha** [comment]
Hi @martyphee i can replicate your issue if i try to install the gem under 2.3. in 2.2 seems to work fine for me. This is leaving me a bit puzzled. Will try to investigate this further.

**Comment by pedrocunha** [comment]
Hi @martyphee please give it a try again with version 1.0.2. Seems the problem was related to the gem being compiled with ruby 2.3.

**Pull request merged**